bothan-coingecko/src/api/builder.rs

```diff
@@ -84,7 +105,7 @@ impl Default for CoinGeckoRestAPIBuilder {
     /// Creates a default `CoinGeckoRestAPIBuilder` instance with default values.
     fn default() -> Self {
         CoinGeckoRestAPIBuilder {
-            url: None,
+            url: Some(DEFAULT_URL.into()),
```

Comment:
So we expect the user with api_key to change the url themself right?

Reply:
default is without an api key, if the user supplies an api key it will auto-change the url to the pro endpoint if it hasn't been set by the user. See line 79-84 for reference

Reply:
Yep, but doesn't put value of some here contradict that purpose, because if user call default function and then put an api key, it will enter case of line 80 which is not a pro url.

Reply:
oh right, it should be None, fixed